core/base-service/metric-helper.js

```diff
@@ -11,9 +11,17 @@ class MetricHelper {
         serviceFamily,
         name,
       })
+      this.serviceResponseSizeHistogram = metricInstance.createServiceResponseSizeHistogram(
```

Comment:
I'm definitely out of my element here on the promethus content, and this is not a blocker for me by any means, more some general questions/food for thought.

I find myself wondering if would it be possible to use more generic names in here (i.e. not including the `Histogram` suffix on `this.serviceResponseSizeHistogram`) and/or if perhaps aspects of the `prom-client` API could be more fully encapsulated within `prometheus-metrics.js`? 

The theoretical use cases I was thinking of (and def not suggesting these) were around what changes would be required here if for example there was an API change `prom-client` (such as `observe(..)` being renamed to `note(..)` in a new major version), if `serviceResponseSize` was changed to a different metric type in `prometheus-metrics.js`, etc.

Reply:
> I find myself wondering if would it be possible to use more generic names in here (i.e. not including the `Histogram` suffix on `this.serviceResponseSizeHistogram`) and/or if perhaps aspects of the `prom-client` API could be more fully encapsulated within `prometheus-metrics.js`?
>
I agree that encapsulating is a good idea. I would like to compare our code with another metrics API which I'm familiar with - Java library Metrics (https://metrics.dropwizard.io/4.1.2/getting-started.html). Metrics uses similar concepts like Prometheus (https://prometheus.io/docs/concepts/metric_types/): counters, gauges, histograms. Name `serviceResponseSizeHistogram` give an info what kind of metric it is and what kind of data it stores. In my opinion we should leave `serviceResponseSizeHistogram` and `serviceRequestCounter` unchanged. 
 
> The theoretical use cases I was thinking of (and def not suggesting these) were around what changes would be required here if for example there was an API change `prom-client` (such as `observe(..)` being renamed to `note(..)` in a new major version), if `serviceResponseSize` was changed to a different metric type in `prometheus-metrics.js`, etc.

Prometheus's counter and Metrics's counter uses `#inc()`. Prometheus's histogram uses `#observe()` and Metrics's histogram uses `#update()` for updates. I just want to show that despite using Prometheus API directly in `metric-helper.js` method we are calling have quite general an universal names. 
Moreover `prometheus-metrics.js` is used only in `metric-helper.js` and this limits possible places requiring changes. 

What do you think about it? :-)

Reply:
> What do you think about it? :-)

I still think that some refactoring could be beneficial, but definitely agreed that everything metric-related is well confined within `core` and that's why I think this is more of a potential follow up (if anything) to this PR. 

My main observation is that when viewing `metric-helper.js` directly, there seem to be some assumptions about the internals of things within `prometheus-metrics.js`. That for me is an indicator that there's the potential opportunity to encapsulate more within `prometheus-metrics.js`

Reply:
OK, I will try to do some refactoring in a separate PR :-)